test/elixir_bench/benchmarks/benchmarks_test.exs

```diff
@@ -1,18 +1,158 @@
+  end
+
+  describe "fetch_benchmark/2" do
+    test "return benchmark given valid params" do
+      my_bench = insert(:benchmark)
+      {:ok, bench} = Benchmarks.fetch_benchmark(my_bench.repo.id, my_bench.name)
+
+      assert_map_attr(my_bench, bench, [:id, :name, :created_at])
+    end
+
+    test "return error if benchmark or repo not found" do
+      repo = insert(:repo)
+
+      assert {:error, :not_found} = Benchmarks.fetch_benchmark(repo.id, "")
+      assert {:error, :not_found} = Benchmarks.fetch_benchmark(nil, "")
+    end
+  end
+
+  describe "fetch_measurement/1" do
+    test "return measurement given valid id" do
+      my_measurement = insert(:measurement)
+      {:ok, measurement} = Benchmarks.fetch_measurement(my_measurement.id)
+
+      # FIXME find a way to also compare job and benchmark for given measurement
+      assert_map_attr(my_measurement, measurement, [:id, :sample_size, :run_times])
+
+      # This does not work because in `measurement` the attributes job and benchmark
+      # are not loaded `#Ecto.Association.NotLoaded`
+      # assert_map_attr(my_measurement, measurement, [:id, :job, :benchmark])
+    end
+
+    test "return error if measurement not found" do
+      assert {:error, :not_found} = Benchmarks.fetch_measurement(nil)
+      assert {:error, :not_found} = Benchmarks.fetch_measurement(9999)
+    end
+  end
+
+  describe "fetch_job/1" do
+    test "return job given valid id" do
+      my_job = insert(:job)
+      {:ok, job} = Benchmarks.fetch_job(my_job.id)
+
+      assert_map_attr(my_job, job, [:id, :branch_name, :commit_message])
+    end
+
+    test "return error if job not found" do
+      assert {:error, :not_found} = Benchmarks.fetch_job(nil)
+      assert {:error, :not_found} = Benchmarks.fetch_job(9999)
+    end
+  end
+
+  describe "fetch_job_by_uuid/1" do
+    test "return job given valid uuid" do
+      my_job = insert(:job)
+      {:ok, job} = Benchmarks.fetch_job_by_uuid(my_job.uuid)
+
+      assert_map_attr(my_job, job, [:id, :uuid, :branch_name, :commit_message])
+    end
+
+    test "return error if job not found" do
+      binary_uuid = "99999999-9999-9999-9999-999999999999"
+      assert {:error, :not_found} = Benchmarks.fetch_job_by_uuid(nil)
+      assert {:error, :not_found} = Benchmarks.fetch_job_by_uuid(binary_uuid)
+    end
+  end
 
   describe "create_job/2" do
-    test "return a new Job given correct params" do
-      {:ok, repo} = Repos.create_repo(%{owner: "elixir-ecto", name: "ecto"})
-      job_attrs = %{branch_name: "mm/benche", commit_sha: "ABC123"}
+    test "insert new Job given correct params" do
+      repo = insert(:repo)
+
+      assert_difference(Job, 1) do
+        Benchmarks.create_job(repo, %{branch_name: "mm/benche", commit_sha: "ABC123"})
+      end
+    end
+
+    test "return the new Job inserted" do
+      repo = insert(:repo)
+
+      {:ok, job} = Benchmarks.create_job(repo, %{branch_name: "mm/benche", commit_sha: "ABC123"})
+      assert %Job{} = job
```

Comment:
I don't think we need both the it's inserted test and it returned it test - both can be rolled into one test imo. Here we could also assert some attributes :)

Reply:
:+1: